outport/process/outportDataProvider.go

```diff
@@ -150,41 +175,201 @@ func (odp *outportDataProvider) getSignersIndexes(header data.HeaderHandler) ([]
+
+	for txHash, txHandler := range txs {
+		tx, castOk := txHandler.(*transaction.Transaction)
+		txHashHex := getHexEncodedHash(txHash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastTransaction, txHashHex)
+		}
+
+		ret[txHashHex] = &outportcore.TxInfo{
+			Transaction: tx,
+			FeeInfo:     newFeeInfo(),
+		}
 	}
+
+	return ret, nil
+}
+
+func getHexEncodedHash(txHash string) string {
+	txHashBytes := []byte(txHash)
+	return hex.EncodeToString(txHashBytes)
 }
 
-func (odp *outportDataProvider) createPoolForMeta(rewardsTxs map[string]data.TransactionHandler) *outportcore.Pool {
-	return &outportcore.Pool{
-		Txs:     WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock)),
-		Scrs:    WrapTxsMap(odp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock)),
-		Rewards: WrapTxsMap(rewardsTxs),
-		Logs:    odp.txCoordinator.GetAllCurrentLogs(),
+func newFeeInfo() *outportcore.FeeInfo {
+	return &outportcore.FeeInfo{
+		GasUsed:        0,
+		Fee:            big.NewInt(0),
+		InitialPaidFee: big.NewInt(0),
 	}
 }
 
-func WrapTxsMap(txs map[string]data.TransactionHandler) map[string]data.TransactionHandlerWithGasUsedAndFee {
-	newMap := make(map[string]data.TransactionHandlerWithGasUsedAndFee, len(txs))
-	for txHash, tx := range txs {
-		newMap[txHash] = outportcore.NewTransactionHandlerWithGasAndFee(tx, 0, big.NewInt(0))
+func getScrs(scrs map[string]data.TransactionHandler) (map[string]*outportcore.SCRInfo, error) {
+	ret := make(map[string]*outportcore.SCRInfo, len(scrs))
+
+	for scrHash, txHandler := range scrs {
+		scr, castOk := txHandler.(*smartContractResult.SmartContractResult)
+		scrHashHex := getHexEncodedHash(scrHash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastSCR, scrHashHex)
+		}
+
+		ret[scrHashHex] = &outportcore.SCRInfo{
+			SmartContractResult: scr,
+			FeeInfo:             newFeeInfo(),
+		}
+	}
+
+	return ret, nil
+}
+
+func getRewards(rewards map[string]data.TransactionHandler) (map[string]*outportcore.RewardInfo, error) {
+	ret := make(map[string]*outportcore.RewardInfo, len(rewards))
+
+	for hash, txHandler := range rewards {
+		reward, castOk := txHandler.(*rewardTx.RewardTx)
+		hexHex := getHexEncodedHash(hash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastReward, hexHex)
+		}
+
+		ret[hexHex] = &outportcore.RewardInfo{
+			Reward: reward,
+		}
+	}
+
+	return ret, nil
+}
+
+func getReceipts(receipts map[string]data.TransactionHandler) (map[string]*receipt.Receipt, error) {
+	ret := make(map[string]*receipt.Receipt, len(receipts))
+
+	for hash, receiptHandler := range receipts {
+		tx, castOk := receiptHandler.(*receipt.Receipt)
+		hashHex := getHexEncodedHash(hash)
+		if !castOk {
+			return nil, fmt.Errorf("%w, hash: %s", errCannotCastReceipt, hashHex)
+		}
+
+		ret[hashHex] = tx
+	}
+
+	return ret, nil
+}
+
+func getLogs(logs []*data.LogData) (map[string]*transaction.Log, error) {
```

Comment:
we change now from somehow ordered logs events to unordered, i'm thinking if this has an impact on the integrators, also for @miiu96

Reply:
yes, logs have to be sorted.

Reply:
Very good catch, fixed it